### PR TITLE
fix: bug missing MapView after importing Map

### PR DESCRIPTION
**Pull request by vietnguyen**
https://jira.dhis2.org/browse/DHIS2-7225

MapView is implementing 3 interfaces `EventAnalyticalObject, MetadataObject, EmbeddedObject`. Those interface have their own logics and are not compatible with each others.

This fix is to add some exception logics to solve those issues.

### CODE REVIEW

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/schema/Property.java [hunk 1]
```diff
@@ -170,6 +170,11 @@
      */
     private boolean embeddedObject;
 
+    /**
+     * Does this class implement {@link EmbeddedObject} ?
+     */
+    private boolean analyticObject;
```
Comment (larshelge) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/schema/Property.java]: Rename to "analyticalObject"

File: dhis-2/dhis-services/dhis-service-schema/src/main/java/org/hisp/dhis/schema/Schema.java [hunk 2]
```diff
@@ -211,6 +211,11 @@
      */
     private Map<String, Property> embeddedObjectProperties;
 
+    /**
+     * Map of all analytic object properties, cached on first request.
+     */
+    private Map<String, Property> analyticObjectProperties;
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-schema/src/main/java/org/hisp/dhis/schema/Schema.java]: Rename to "analyticalObjectProperties"

File: dhis-2/dhis-services/dhis-service-schema/src/main/java/org/hisp/dhis/schema/Schema.java [hunk 3]
```diff
@@ -605,6 +610,20 @@ public void setPropertyMap( Map<String, Property> propertyMap )
         return embeddedObjectProperties;
     }
 
+    public Map<String, Property> getAnalyticObjectProperties()
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-schema/src/main/java/org/hisp/dhis/schema/Schema.java]: Rename to "getAnalyticalObjectProperties"

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/schema/Property.java (hunk 1 above)
Comment (vietnguyen) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/schema/Property.java]: fixed

File: dhis-2/dhis-services/dhis-service-schema/src/main/java/org/hisp/dhis/schema/Schema.java (hunk 2 above)
Comment (vietnguyen) [reply dhis-2/dhis-services/dhis-service-schema/src/main/java/org/hisp/dhis/schema/Schema.java]: fixed

File: dhis-2/dhis-services/dhis-service-schema/src/main/java/org/hisp/dhis/schema/Schema.java (hunk 3 above)
Comment (vietnguyen) [reply dhis-2/dhis-services/dhis-service-schema/src/main/java/org/hisp/dhis/schema/Schema.java]: fixed

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/AnalyticalObjectImportHandler.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package org.hisp.dhis.dxf2.metadata;
+
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/AnalyticalObjectImportHandler.java]: Add license.

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/DefaultAnalyticalObjectImportHandler.java [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+            if ( dataDimensionItem.getDataElementOperand() != null )
+            {
+                preheatService.connectReferences( dataDimensionItem.getDataElementOperand(), bundle.getPreheat(), bundle.getPreheatIdentifier() );
+                session.save( dataDimensionItem.getDataElementOperand() );
+            }
+
+            if ( dataDimensionItem.getReportingRate() != null )
+            {
+                dataDimensionItem.getReportingRate().setDataSet( bundle.getPreheat().get( bundle.getPreheatIdentifier(),
+                        dataDimensionItem.getReportingRate().getDataSet() ) );
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/DefaultAnalyticalObjectImportHandler.java]: Use 4 space indent.

**Pull request merged**